luigi/task.py

```diff
@@ -363,7 +388,15 @@ def get_params(cls):
             param_obj = getattr(cls, param_name)
             if not isinstance(param_obj, Parameter):
                 continue
+            params.append((param_name, param_obj))
 
+        pep593_params = _get_pep539_annotations(cls)
+        for param_name, param_obj in pep593_params:
+            try:
+                default_val = getattr(cls, param_name)
+                param_obj._default = default_val
+            except AttributeError:  # no default value
+                pass
```

Comment:
Should we notify the user, e.g. as a warning, that something went wring and the parameter won't be used?